transform-plugins/src/main/java/io/cdap/plugin/ValueMapper.java

```diff
@@ -82,6 +87,29 @@ public Config(String mapping, String defaults) {
     }
   }
 
+  @Override
+  public void prepareRun(StageSubmitterContext context) throws Exception {
+    super.prepareRun(context);
+    if (context.getInputSchema() == null || context.getInputSchema().getFields() == null) {
+      return;
+    }
+    parseConfiguration(this.config, context.getFailureCollector());
+
+    // After extracting the mappings, store a list of operations containing identity transforms for every output
+    // field also present in the mappings list.
+    List<String> mappedFields = TransformLineageRecorderUtils.getFields(context.getInputSchema())
+      .stream().filter(mappingValues::containsKey).collect(Collectors.toList());
+
+    List<String> identityFields = TransformLineageRecorderUtils.getFields(context.getInputSchema());
+    identityFields.removeAll(mappedFields);
+
+    List<FieldOperation> output = TransformLineageRecorderUtils.generateOneToOnes(mappedFields, "mapValueOf",
```

Comment:
Here the logic is wrong. The mapped fields actually become field with other name in output schema, see https://github.com/cdapio/hydrator-plugins/blob/8802b9906ebe7cc26cd6fe71f68849cd2931738b/transform-plugins/src/main/java/io/cdap/plugin/ValueMapper.java#L226. Therefore, here it should not assume it is similar to identity transform.